src/main/java/net/mcreator/ui/modgui/ProcedureGUI.java

```diff
@@ -603,12 +603,15 @@ public Component getListCellRendererComponent(JList<? extends VariableElement> l
 
 		// this procedure could be in use and new dependencies were added
 		if (isEditingMode() && dependenciesChanged)
-			regenerateProcedureCallers(modElement, modElement);
+			regenerateProcedureCallers(modElement, new Stack<>());
 
 		dependenciesBeforeEdit = dependenciesArrayList;
 	}
 
-	private void regenerateProcedureCallers(ModElement procedure, ModElement recursionLock) {
+	private void regenerateProcedureCallers(ModElement procedure, Stack<ModElement> recursionLock) {
+		if (recursionLock.contains(procedure))

```

Comment:
I would just equip this with a few comments to explain why we check here and why stack so we don't scratch our heads too long in the future :)

Otherwise looks good, nice catch and fix!

Reply:
Like now?